calendar_resource/models/calendar_event.py

```diff
@@ -16,11 +18,89 @@ class CalendarEvent(models.Model):
         comodel_name='resource.resource',
     )
 
-    @api.constrains('resource_ids')
-    def _check_resource_ids_dbl_book(self):
+    @api.model
+    def _format_datetime_interval_list(self, intervals):
+        """ Converts intervals to string values for front-end
+            display purposes, taking into account language,
+            timezone, as well as date and time formats.
+
+        Example:
+
+            .. code-block python
+
+            self._format_datetime_interval_list(
+                intervals=[
+                    (
+                        datetime(2017, 6, 28, 17, 0, 0),
+                        datetime(2017, 6, 29, 8, 0, 0),
+                    ),
+                    (
+                        datetime(2017, 6, 29, 12, 0, 0),
+                        datetime(2017, 6, 29, 13, 0, 0),
+                    ),
+                ],
+            )
+
+            The above example will output the following string,
+            depending on date and time format, as well as timezone.
+
+            \n06/28/2017 at 17:00:00 To\n
+            06/29/2017 at 08:00:00 (UTC)\n
+
+            \n06/29/2017 at 12:00:00 To\n
+            06/29/2017 at 13:00:00 (UTC)\n
+
+        """
+        datetimes = ''
+        for interval in intervals:
+            if not isinstance(interval[0], str):
+                interval = (
+                    fields.Datetime.to_string(interval[0]),
+                    fields.Datetime.to_string(interval[1]),
+                )
+            args = {
+                'start': interval[0],
+                'stop': interval[1],
+                'zallday': False,
+                'zduration': 24,
+            }
+            datetimes += '\n%s\n' % self._get_display_time(**args)
+        return datetimes
+
+    @api.multi
+    def _event_in_past(self):
+        self.ensure_one()
+        stop_datetime = fields.Datetime.from_string(self.stop)
+        now_datetime = fields.Datetime.from_string(fields.Datetime.now())
+        return stop_datetime < now_datetime
+
+    @api.multi
+    def _get_event_date_list(self):
+        self.ensure_one()
+        start = fields.Date.from_string(self.start)
+        stop = fields.Datetime.from_string(self.stop)
+
+        if stop.time() == time(0, 0):
```

Comment:
No all of that is done in the remove_interval_time_overlaps method and the round_up method in resource.calendar. 

This just ensures that if the stop time ends at the beginning of a day, not to include that day. When I loop through all the days of the event, if I included that day then all the working times for that day would be unnecessarily retrieved.

I originally tried working with rrule, but found there to be too many edge cases for me to work with it, so I built this custom method.

Now that the edge cases have been ironed out, it seems I could just forego this custom method altogether, and simply remove the last date from the list if this edge case is true.